frontend/src/metabase/visualizations/echarts/cartesian/chart-measurements/index.ts

```diff
@@ -161,12 +235,41 @@ export const getChartPadding = (
   const yAxisNameTotalWidth =
     CHART_STYLE.axisName.size / 2 + CHART_STYLE.axisNameMargin;
 
-  if (chartModel.leftAxisModel?.label) {
-    padding.left += yAxisNameTotalWidth;
-  }
-  if (chartModel.rightAxisModel?.label) {
-    padding.right += yAxisNameTotalWidth;
-  }
+  const currentBoundaryWidth =
+    chartWidth -
+    padding.left -
+    padding.right -
+    ticksDimensions.yTicksWidthLeft -
+    ticksDimensions.yTicksWidthRight;
+  const dimensionWidth = currentBoundaryWidth / chartModel.dataset.length;
+  const firstTickPosition = dimensionWidth / 2;
+  const lastTickPosition = currentBoundaryWidth - dimensionWidth / 2;
```

Comment:
Since we dont't really need ticks positions within the boundary I found this a bit confusing.
For category scales ticks are located at the center of each category segment so I guess it can be
```
const categoryTickOffset = dimensionWidth / 2
let firstTickOverflow = firstTickWidth / 2
let lastTickOverflow = lastTickWidth / 2

// on category scale the tick center is in the middle of the category segment
// while on time scale tick can center can be right on the edge
if (isOrdinal) {
  firstTickOverflow -= categoryTickOffset
  lastTickOverflow -= categoryTickOffset
}

// apply limiting of huge tick overflows
``

Reply:
Sure I can remove the position.

> // while on time scale tick can center can be right on the edge

Can you please share an example of this happening? On every timeseries chart I look at the label is centered over its datum

<img width="752" alt="Screenshot 2024-02-23 at 10 18 21 AM" src="https://github.com/metabase/metabase/assets/37751258/0f5a757f-f6a9-44d7-ad41-cf6a0eef6ee7">